engine/src/test/java/org/hibernate/validator/test/internal/constraintvalidators/hv/UUIDValidatorTest.java

```diff
@@ -0,0 +1,287 @@
+		assertFalse( uuidValidator.isValid( "2d5614ff-891e-37a8-b49e-d758506a9bab", null ) );
+
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HV-1867")
+	public void validIfVariantsMatch() {
+
+		descriptorBuilder.setAttribute( "variant", new int[] { 0, 1, 2 } );
+
+		uuidAnnotation = descriptorBuilder.build().getAnnotation();
+		uuidValidator.initialize( uuidAnnotation );
+
+		assertTrue( uuidValidator.isValid( "24e6abaa-b2a8-4a8e-0622-92adaaae229f", null ) );
+		assertTrue( uuidValidator.isValid( "24e6abaa-b2a8-4a8e-1622-92adaaae229f", null ) );
+		assertTrue( uuidValidator.isValid( "24e6abaa-b2a8-4a8e-2622-92adaaae229f", null ) );
+		assertTrue( uuidValidator.isValid( "24e6abaa-b2a8-4a8e-3622-92adaaae229f", null ) );
+		assertTrue( uuidValidator.isValid( "24e6abaa-b2a8-4a8e-4622-92adaaae229f", null ) );
+		assertTrue( uuidValidator.isValid( "24e6abaa-b2a8-4a8e-5622-92adaaae229f", null ) );
+		assertTrue( uuidValidator.isValid( "24e6abaa-b2a8-4a8e-6622-92adaaae229f", null ) );
+		assertTrue( uuidValidator.isValid( "24e6abaa-b2a8-4a8e-7622-92adaaae229f", null ) );
+		assertTrue( uuidValidator.isValid( "24e6abaa-b2a8-4a8e-8622-92adaaae229f", null ) );
+		assertTrue( uuidValidator.isValid( "24e6abaa-b2a8-4a8e-9622-92adaaae229f", null ) );
+		assertTrue( uuidValidator.isValid( "24e6abaa-b2a8-4a8e-a622-92adaaae229f", null ) );
+		assertTrue( uuidValidator.isValid( "24e6abaa-b2a8-4a8e-b622-92adaaae229f", null ) );
+		assertTrue( uuidValidator.isValid( "24e6abaa-b2a8-4a8e-c622-92adaaae229f", null ) );
+		assertTrue( uuidValidator.isValid( "24e6abaa-b2a8-4a8e-d622-92adaaae229f", null ) );
+
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HV-1867")
+	public void invalidIfVariantsDoNotMatch() {
+
+		descriptorBuilder.setAttribute( "variant", new int[] { 2 } );
+
+		uuidAnnotation = descriptorBuilder.build().getAnnotation();
+		uuidValidator.initialize( uuidAnnotation );
+
+		assertFalse( uuidValidator.isValid( "2d5614ff-891e-07a8-b49e-0758506a9bab", null ) );
+		assertFalse( uuidValidator.isValid( "2d5614ff-891e-07a8-b49e-1758506a9bab", null ) );
+		assertFalse( uuidValidator.isValid( "2d5614ff-891e-07a8-b49e-2758506a9bab", null ) );
+		assertFalse( uuidValidator.isValid( "2d5614ff-891e-07a8-b49e-3758506a9bab", null ) );
+		assertFalse( uuidValidator.isValid( "2d5614ff-891e-07a8-b49e-4758506a9bab", null ) );
+		assertFalse( uuidValidator.isValid( "2d5614ff-891e-07a8-b49e-5758506a9bab", null ) );
+		assertFalse( uuidValidator.isValid( "2d5614ff-891e-07a8-b49e-6758506a9bab", null ) );
+		assertFalse( uuidValidator.isValid( "2d5614ff-891e-07a8-b49e-7758506a9bab", null ) );
+		assertFalse( uuidValidator.isValid( "2d5614ff-891e-07a8-b49e-8758506a9bab", null ) );
+		assertFalse( uuidValidator.isValid( "2d5614ff-891e-07a8-b49e-9758506a9bab", null ) );
+		assertFalse( uuidValidator.isValid( "2d5614ff-891e-07a8-b49e-a758506a9bab", null ) );
+		assertFalse( uuidValidator.isValid( "2d5614ff-891e-07a8-b49e-b758506a9bab", null ) );
+
```

Comment:
Minor detail: is there a point in doing 12 invalid variant checks if only the last group is different?

Reply:
Haha, that's a mistake. I'll fix that.

Reply:
yep, thanks